ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -234,10 +234,16 @@ def __init__(
         super().__init__()
         self.action_spec = action_spec
         self.version_number = torch.nn.Parameter(torch.Tensor([2.0]))
-        self.is_continuous_int = torch.nn.Parameter(
+        self.is_continuous_int_deprecated = torch.nn.Parameter(
             torch.Tensor([int(self.action_spec.is_continuous())])
         )
-        self.act_size_vector = torch.nn.Parameter(
+        self.continuous_act_size_vector = torch.nn.Parameter(
+            torch.Tensor([int(self.action_spec.continuous_size)]), requires_grad=False
+        )
+        self.discrete_act_size_vector = torch.nn.Parameter(
+            torch.Tensor([sum(self.action_spec.discrete_branches)]), requires_grad=False
```

Comment:
I wonder if the sum is really what we want. To make sure the model is compatible we would need all of the discrete branches, or only the number of branches if we want to make sure the action tensor has the right shape. I do not see how the sum will be useful at inference.

Reply:
Ideally, at inference, we will no longer use our own C# multinomial function to do the sampling and the sampling can happen in the ONNX model

Reply:
Agreed, but here I followed the old model output to keep the output same as before.
I can change this to list of branch sizes but then the new model will be exporting different things than the old model.

I would prefer doing this change later when we make sure things are working and also when we migrate ActionSpec with BrainParameters in C# so that I can actually do the check that every branch size matches.

Reply:
On one hand, changing the model output format is a big scope creep for this (already large) change.

On the other hand, since we're already changing the output tensor names, this _could_ be a time to change the format without introducing extra complexity/versioning on the C# side.

I don't have a strong preference either way; maybe wait until everything is working on staging before making a decision?

Reply:
Left a todo here for now